### PR TITLE
Fusion optimization now able to return regular patterns

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/adblock-rust/issues/255

image

### CODE REVIEW

File: src/optimizer.rs [hunk 1]
```diff
@@ -222,15 +221,19 @@ mod optimization_tests_pattern_group {
     fn check_match(
```
Comment (atuchin-m) [inline src/optimizer.rs:221]: check_match should take into account the filter could be non-regex.

File: src/optimizer.rs [hunk 2]
```diff
@@ -284,7 +287,7 @@ mod optimization_tests_pattern_group {
 
         let fused = optimization.fusion(&filters);
 
-        assert!(fused.is_regex(), "Expected rule to be regex");
+        assert!(fused.is_regex() == false, "Expected rule to not be a regex");
```
Comment (atuchin-m) [inline src/optimizer.rs:290]: Now it checks non-regex case. We could also add another case if needed.

File: src/optimizer.rs [hunk 3]
```diff
@@ -102,8 +102,7 @@ impl Optimization for SimplePatternGroup {
             }
         }
 
-        // let is_regex = filters.iter().find(|f| f.is_regex()).is_some();
```
Comment (atuchin-m) [inline src/optimizer.rs:105]: Any idea why it was disabled?
I also believe we shouldn't fuse regex with non regex rules.
So a possible solution is returning a pair or a vector `[fused_non_regex_rules, fused_regex_rules]`

Comment (antonok-edm) [reply src/optimizer.rs:105]: https://github.com/brave/adblock-rust/commit/5c28aefc#diff-40a35a91a642ade2cf6b8ce86ddb3c8b82625c0592d0e9fa24d2b41a072ab3adR92

No idea :stuck_out_tongue:

File: src/optimizer.rs (hunk 2 above)
Comment (antonok-edm) [reply src/optimizer.rs:290]: Sounds like there are gonna be some changes here anyways so this should be fine for now

**Review by antonok-edm — APPROVED** [review]

**Pull request merged**